packages/rich-text-editor/src/styles/vaadin-rich-text-editor-base-styles.js

```diff
@@ -0,0 +1,352 @@
+
+  .ql-editor > * {
+    cursor: text;
+  }
+
+  .ql-align-left {
+    text-align: left;
+  }
+
+  .ql-direction-rtl {
+    direction: rtl;
+    text-align: inherit;
+  }
+
+  .ql-align-center {
+    text-align: center;
+  }
+
+  .ql-align-justify {
+    text-align: justify;
+  }
+
+  .ql-align-right {
+    text-align: right;
+  }
+  /* quill core end */
+
+  blockquote {
+    border-inline-start: 4px solid var(--vaadin-border-color);
+    margin: var(--vaadin-padding-container);
+    padding-inline-start: var(--vaadin-padding);
+  }
+
+  code,
+  pre {
+    background-color: var(--vaadin-background-container);
+    border-radius: var(--vaadin-radius-s);
+  }
+
+  pre {
+    white-space: pre-wrap;
+    margin-block: var(--vaadin-padding);
+    padding: var(--vaadin-padding-container);
+  }
+
+  code {
+    padding: 0.125rem 0.25rem;
+  }
+
+  img {
+    max-width: 100%;
+  }
+
+  /* RTL specific styles */
+  :host([dir='rtl']) .ql-editor {
+    direction: rtl;
+    text-align: right;
+  }
+`;
+
+const toolbar = css`
+  [part='toolbar'] {
+    background-color: var(--vaadin-rte-toolbar-background, var(--vaadin-background-container));
+    display: flex;
+    flex-shrink: 0;
+    flex-wrap: wrap;
+    gap: var(--vaadin-rte-toolbar-gap, var(--vaadin-gap-container-inline));
+    padding: var(--vaadin-rte-toolbar-padding, var(--vaadin-padding));
+  }
+
+  [part~='toolbar-group'] {
+    display: flex;
+  }
+
+  [part~='toolbar-button'] {
+    background: var(--vaadin-rte-toolbar-button-background, var(--vaadin-background-container));
+    border: var(
+      --vaadin-rte-toolbar-button-border,
+      var(--vaadin-rte-toolbar-button-border-width, 1px) solid
+        var(--vaadin-rte-toolbar-button-border-color, transparent)
+    );
+    border-radius: var(--vaadin-rte-toolbar-button-border-radius, var(--vaadin-radius-m));
+    color: var(--vaadin-rte-toolbar-button-text-color, var(--vaadin-color));
+    cursor: var(--vaadin-clickable-cursor);
+    flex-shrink: 0;
+    font-family: var(--vaadin-rte-toolbar-button-font-family, inherit);
+    font-size: var(--vaadin-rte-toolbar-button-font-size, inherit);
+    font-weight: var(--vaadin-rte-toolbar-button-font-weight, 500);
+    height: var(--vaadin-rte-toolbar-button-height, auto);
+    line-height: var(--vaadin-rte-toolbar-button-line-height, inherit);
+    padding: var(--vaadin-rte-toolbar-button-padding, var(--vaadin-padding-container));
+    position: relative;
+  }
+
+  [part~='toolbar-button']::before {
+    background: currentcolor;
+    content: '';
+    display: block;
+    height: var(--vaadin-icon-size, 1lh);
+    width: var(--vaadin-icon-size, 1lh);
+  }
+
+  [part~='toolbar-button']:focus {
+    outline: var(--vaadin-focus-ring-width) solid var(--vaadin-focus-ring-color);
+    outline-offset: 1px;
+    z-index: 1;
+  }
+
+  [part~='toolbar-button'][on],
+  [part~='toolbar-button'][aria-expanded='true'] {
+    --vaadin-rte-toolbar-button-background: var(--vaadin-background-container-strong);
```

Comment:
Could benefit from a `--vaadin-rte-toolbar-button-on-background`. But that begs the question if border and text colors should also get such properties. So I’m thinking let’s skip it for now, and see how this feels with Aura.